### DISCUSSION TITLE
Use GitHub Flavored Markdown for README rendering

### DISCUSSION THREAD

**Discussion by ethomson** [npm/feedback#3] [Website] [answered]
https://npmjs.com renders `README.md` files that are in packages with markdown.  _However_ the markdown parsing is a bit...  antiquated.  It would be great to use [GitHub-flavored markdown](https://github.github.com/gfm/) to support things like tables, and a consistent experience between viewing READMEs on GitHub vs on the package page on npmjs.com.

**Comment by MylesBorins** [comment, marked as answer]
We are now tracking this feature on the roadmap

https://github.com/npm/roadmap/issues/15

**Reply by lumaxis** [reply]
Adding a link to the specific roadmap item: https://github.com/npm/roadmap/issues/15

**Reply by bnb** [reply]
I cannot explain how relieving this is.

**Comment by wooorm** [comment]
I have recently done a lot of reverse engineering into how GH renders markdown (which is somewhat different than GFM / cmark-gfm) to match exactly how it works on github.com for [remark](https://github.com/remarkjs/remark).

Also: @zeke made the original [marky-markdown](https://www.npmjs.com/package/@npmcorp/marky-markdown), now works at GH, and has recently used remark to build GHs help docs.

**Reply by ethomson** [reply]
Thanks @wooorm, I'll pass this along to the engineering team.

**Reply by zeke** [reply]
It me.

Catching up on notifications. 😅  Happy to lend a hand to this effort if needed.

**Comment by octref** [comment]
Somewhat related, can you make npmjs.com to not render `readme.txt` as Markdown while you are on it? I don't think this should be a complex change.

My package with only `readme.txt` but not `readme.md`: https://github.com/octref/arvelie

## GitHub

![image](https://user-images.githubusercontent.com/4033249/97298450-5f67e880-188e-11eb-9c58-93c283476382.png)

## NPM

![image](https://user-images.githubusercontent.com/4033249/97298470-67c02380-188e-11eb-97f4-9554480b64fd.png)

**Reply by MylesBorins** [reply]
We'll make sure that we take steps to think about various formats folks use for READMEs when designing this feature

**Reply by ethomson** [reply]
@octref we have fixed this - `README.txt` files are now treated as text, not markdown.  https://www.npmjs.com/package/arvelie

**Reply by ethomson** [reply]
(It looks like we need to render that in a monospace font.)

**Reply by octref** [reply]
@ethomson Thanks a lot for fixing it! IBM Plex is free and looks pretty good. But if you want to avoid adding more assets, here's a popular monospace stack: https://www.digitalocean.com/community/tutorials/css-system-font-stack
